src/core/src/EnvironmentOptions.C

```diff
@@ -46,139 +46,24 @@ namespace QUESO {
 // Default constructor ------------------------------
 EnvOptionsValues::EnvOptionsValues()
   :
-    m_prefix("env_"),
-    m_help(UQ_ENV_HELP),
-    m_numSubEnvironments(UQ_ENV_NUM_SUB_ENVIRONMENTS_ODV),
-    m_subDisplayFileName(UQ_ENV_SUB_DISPLAY_FILE_NAME_ODV),
-    m_subDisplayAllowAll(UQ_ENV_SUB_DISPLAY_ALLOW_ALL_ODV),
-    m_subDisplayAllowInter0(UQ_ENV_SUB_DISPLAY_ALLOW_INTER0_ODV),
-    //m_subDisplayAllowedSet (),
-    m_displayVerbosity(UQ_ENV_DISPLAY_VERBOSITY_ODV),
-    m_syncVerbosity(UQ_ENV_SYNC_VERBOSITY_ODV),
-    m_checkingLevel(UQ_ENV_CHECKING_LEVEL_ODV),
-    m_rngType(UQ_ENV_RNG_TYPE_ODV),
-    m_seed(UQ_ENV_SEED_ODV),
-    m_platformName(UQ_ENV_PLATFORM_NAME_ODV),
-    m_identifyingString(UQ_ENV_IDENTIFYING_STRING_ODV),
-    m_numDebugParams(UQ_ENV_NUM_DEBUG_PARAMS_ODV),
-    m_debugParams(m_numDebugParams,0.),
-    m_env(NULL),
+  m_env(NULL)
 #ifndef QUESO_DISABLE_BOOST_PROGRAM_OPTIONS
-    m_parser(),
+  , m_parser(new BoostInputOptionsParser())
```

Comment:
Why do we default construct a BoostInputOptionsParser for the empty input file case?

Reply:
Wait, isn't the ```new BoostInputOptionsParser()``` bit just in your patch?  The previous code was just ```m_parser()```, leaving it an uninitialized scoped pointer.

Reply:
Yes, you're right.  I caused confusion here by not making reference and relaying a question I was more asking myself.  I took a cue from https://github.com/libqueso/queso/blob/c6c71a0ee2bc81c7316fe67dbff104dc8104b64e/src/gp/src/GPMSAOptions.C#L133 where it gets initialized, but I agree leaving the scoped ptr uninitialized should be okay.

Reply:
Ah, that looks like a mistake on my part.  If we need m_parser then we reset it anyway, so there's no point in initializing it in the first place...

Reply:
@roystgnr  FWIW, I learned the hard way why this initialization is necessary: operator<< dereferences this object, which would fail after default construction.  I added initialization to all classes for now and will promptly remove it post 0.58.0.  Same bug could affect copy ctors, but that use case isn't exercised by any code right now, so ignoring.